### PR TITLE
WIP: Move to Quay.io

### PR DESCRIPTION
**Pull request by jmelis**
This PR is part of the effort to move to Quay.

This is the list of changes in this PR:

- Pushes to Quay instead of the Devshift registry
- Uses an image hosted in Quay to build the RHEL based container image
- Uses the new env-toolkit to load variables from jenkins-env
- Changes the default path of the image to quay (note that this should not affect production because it is overridden in the saas repo)

A companion PR should have been submitted to the appropriate saas repo to change the staging url from devshift to quay. The PR to the saas repo should be merged before this one.

### CODE REVIEW

**Comment by jmelis** [comment]
@ibuziuk  @davidfestal  (cc @riuvshin ) I'm a bit stuck with this PR, can you guys help me out?

The ultimate goal is to deprecate the use of registry.devshift and replace it with quay.io. Most projects push a single image to the registry, but I found that this projects pushes several ones.

If we look at the catalog [https://registry.devshift.net/v2/_catalog](https://gist.github.com/jmelis/70c5b7f0147c30f8bd887c6e07ae98d9) we can see these images:

- `che/rh-che-server`
- `che/rh-che-standalone-keycloak-configurator`
- `osio-prod/rhchestage/rh-che-automation`
- `osio-prod/rhchestage/rh-che-server`
- `osio-prod/rhchestage/rh-che-standalone-keycloak-configurator`
- `rhchestage/rh-che-server`
- `rhchestage/rh-che-standalone-keycloak-configurator`

And it looks like they have been recently created:

```bash
$ skopeo inspect docker://registry.devshift.net/che/rh-che-standalone-keycloak-configurator | jq -r .Created
2018-06-27T16:45:17.996443132Z
$ skopeo inspect docker://registry.devshift.net/che/rh-che-server | jq -r .Created      
2018-06-27T16:43:41.547486127Z
```

But I don't understand how this is happening. I don't undertand how the `che/rh-che-standalone-keycloak-configurator` is being pushed.

```
.ci/cico_do_docker_build_tag_push.sh:92:          docker push ${REGISTRY}/${NAMESPACE}/${KEYCLOAK_STANDALONE_CONFIGURATOR_IMAGE}:${TAG}
```

But `$NAMESPACE` is **only** defined here:

```
config:6:export NAMESPACE=${NAMESPACE:-"rhchestage"}
```

How come it's being pushed to `che/rh-che-standalone-keycloak-configurator` instead of `rhchestage/rh-che-standalone-keycloak-configurator`?

Another thing I don't understand is why there is `osio-prod/rhchestage/rh-che-automation` but there isn't `rhchestage/rh-che-automation`? Which ones are we really using?

**Comment by davidfestal** [comment]
> How come it's being pushed to che/rh-che-standalone-keycloak-configurator instead of rhchestage/rh-che-standalone-keycloak-configurator?
> 
> Another thing I don't understand is why there is osio-prod/rhchestage/rh-che-automation but there isn't rhchestage/rh-che-automation? Which ones are we really using?

@jmelis the `.ci/cico_do_docker_build_tag_push.sh` pushed the 2 images to a staging repository. Then the images are pushed to the real production repository after validation and test success, by this script: `.ci/cico_deploy.sh`

**Comment by jmelis** [comment]
@davidfestal thanks a lot!!! I missed that line for some reason!

What about the automatin image? is that being used?

**Comment by jmelis** [comment]
[test]

**Comment by jmelis** [comment]
related PR: https://github.com/openshiftio/saas-openshiftio/pull/956

**Comment by jmelis** [comment]
[test]

**Comment by jmelis** [comment]
[test]

**Comment by jmelis** [comment]
[test]

**Comment by ibuziuk** [comment]
> What about the automation image? is that being used?
@jmelis it is supposed to used for deployment on dev cluster for running the tests against it.
@ScrewTSW could provide more details

**Comment by jmelis** [comment]
@ibuziuk can you help me figure out why this build is failing at the deploy postgres stage?

**Comment by ibuziuk** [comment]
@jmelis sorry a bit of busy right now but might be @ScrewTSW would be able to help. He is the original author of the deployment script to dev cluster. What I would check if the db deployment works in upstream [1] and than check what might cause failures for rh-che (e.g. wrong image etc)

[1] https://github.com/eclipse/che/tree/master/deploy/openshift/templates#deploy-multi-user-che-with-bundled-keycloak-and-postgres-http

**Comment by ScrewTSW** [comment]
@jmelis There is an issue with OSD on `dev.rdu2c.fabric8.io` as the postgres container fails to deploy.
screenshot_20180704_101648
the deploy pod just seems to be stuck, doing nothing until it times out, no logs
image pasted at 2018-7-4 10-18

File: openshift/rh-che.app.yaml [hunk 1]
```diff
@@ -347,6 +347,6 @@ objects:
       name: rhche-host
 parameters:
 - name: IMAGE
-  value: registry.devshift.net/che/rh-che-server
+  value: quay.io/openshiftio/che-rh-che-server
```
Comment (ibuziuk) [inline openshift/rh-che.app.yaml]: @jmelis https://github.com/openshiftio/saas-openshiftio/pull/956/files

It looks like the image in saas-openshiftio is different 
- `quay.io/openshiftio/rhel-che-rh-che-server`
- `quay.io/openshiftio/che-rh-che-server`

Comment (jmelis) [reply openshift/rh-che.app.yaml]: sounds good to me, but I just realised it should be preffixed with `rhel-` to indicate that it's rhel based. making the change, thanks for reviewing!!

**Comment by ibuziuk** [comment]
[test]

**Pull request merged**